diff --git a/scream/components/eamxx/src/physics/p3/p3_iso_c.f90 b/scream/components/eamxx/src/physics/p3/p3_iso_c.f90
--- scream/components/eamxx/src/physics/p3/p3_iso_c.f90
+++ scream/components/eamxx/src/physics/p3/p3_iso_c.f90
@@ -15,9 +15,9 @@
 
 contains
   subroutine append_precision(string, prefix)
 
-    character(kind=c_char, len=256), intent(inout) :: string
+    character(kind=c_char, len=512), intent(out) :: string
     character(*), intent(in) :: prefix
     real(kind=c_real) :: s
 
     write (string, '(a,i1,a1)') prefix, sizeof(s), C_NULL_CHAR
@@ -56,9 +56,9 @@
     real(kind=c_real), dimension(150), target :: mu_r_table_vals
     real(kind=c_real), dimension(300,10), target :: vn_table_vals, vm_table_vals, revap_table_vals
 
     character(len=256), pointer :: lookup_file_dir
-    character(kind=c_char, len=256) :: mu_r_filename, revap_filename, vn_filename, vm_filename
+    character(kind=c_char, len=512) :: mu_r_filename, revap_filename, vn_filename, vm_filename
     integer :: len
     logical :: ok
     character(len=16) :: p3_version="4.1.1"  ! TODO: Change to be dependent on table version and path specified in p3_functions.hpp
 
@@ -68,12 +68,19 @@
 
     info = 0
     ok = .false.
 
-    call append_precision(mu_r_filename, SCREAM_DATA_DIR//"/tables/mu_r_table_vals.dat")
-    call append_precision(revap_filename, SCREAM_DATA_DIR//"/tables/revap_table_vals.dat")
-    call append_precision(vn_filename, SCREAM_DATA_DIR//"/tables/vn_table_vals.dat")
-    call append_precision(vm_filename, SCREAM_DATA_DIR//"/tables/vm_table_vals.dat")
+#ifdef SCREAM_DOUBLE_PRECISION
+    mu_r_filename  = lookup_file_dir(1:len)//'/mu_r_table_vals.dat8'//C_NULL_CHAR
+    revap_filename = lookup_file_dir(1:len)//'/revap_table_vals.dat8'//C_NULL_CHAR
+    vn_filename    = lookup_file_dir(1:len)//'/vn_table_vals.dat8'//C_NULL_CHAR
+    vm_filename    = lookup_file_dir(1:len)//'/vm_table_vals.dat8'//C_NULL_CHAR
+#else
+    mu_r_filename  = lookup_file_dir(1:len)//'/mu_r_table_vals.dat4'//C_NULL_CHAR
+    revap_filename = lookup_file_dir(1:len)//'/revap_table_vals.dat4'//C_NULL_CHAR
+    vn_filename    = lookup_file_dir(1:len)//'/vn_table_vals.dat4'//C_NULL_CHAR
+    vm_filename    = lookup_file_dir(1:len)//'/vm_table_vals.dat4'//C_NULL_CHAR
+#endif
 
     if (write_tables) then
        call p3_init_b()
        call p3_get_tables(mu_r_table_vals, revap_table_vals, vn_table_vals, vm_table_vals)
